toqito/matrix_ops/commutant.py

```diff
@@ -0,0 +1,60 @@
+"""Module for computing the commutant of a set of matrices."""
+
+import numpy as np
+from scipy.linalg import null_space
+
+
+def commutant(A: np.ndarray | list[np.ndarray]) -> list[np.ndarray]:
+    r"""Compute an orthonormal basis for the commutant algebra.
+
+    Given a set of matrices A, this function determines an orthonormal basis 
+    (with respect to the Hilbert-Schmidt inner product) for the algebra of matrices 
+    that commute with every matrix in A.
+
+
+    The commutant condition is given by:
+
+    .. math::
+        A X = X A \quad \forall A \in \mathcal{A}.
+
+    This can be rewritten as:
+
+    .. math::
+        (A \otimes I - I \otimes A^T) \text{vec}(X) = 0.
+
+    where :math:`\text{vec}(X)` denotes the vectorization of :math:`X`. The null space of this
+    equation provides a basis for the commutant.
```

Comment:
Please add an example in the docstring. You can take a look at other functions in the API. 

For example, see https://toqito.readthedocs.io/en/latest/autoapi/matrix_ops/tensor/index.html#matrix_ops.tensor.tensor